workers/go/workflows/throughput_stress.go

```diff
@@ -0,0 +1,181 @@
+	"github.com/temporalio/omes/workers/go/activities"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+const (
+	UpdateSleep         = "sleep"
+	UpdateActivity      = "activity"
+	UpdateLocalActivity = "localActivity"
+	ASignal             = "someSignal"
+)
+
+// ThroughputStressWorkflow is meant to mimic the throughputstress scenario from bench-go of days
+// past, but in a less-opaque way. We do not ask why it is the way it is, it is not our place to
+// question the inscrutable ways of the old code.
+func ThroughputStressWorkflow(ctx workflow.Context, params *througput_stress.WorkflowParams) (string, error) {
+	// Establish handlers
+	err := workflow.SetQueryHandler(ctx, "myquery", func() (string, error) {
+		return "hi", nil
+	})
+	if err != nil {
+		return "", err
+	}
+	err = workflow.SetUpdateHandler(ctx, UpdateSleep, func(ctx workflow.Context) error {
+		return workflow.Sleep(ctx, 1*time.Second)
+	})
+	if err != nil {
+		return "", err
+	}
+	err = workflow.SetUpdateHandler(ctx, UpdateActivity, func(ctx workflow.Context) error {
+		actCtx := workflow.WithActivityOptions(ctx, defaultActivityOpts())
+		return workflow.ExecuteActivity(
+			actCtx, "Payload", activities.MakePayloadInput(0, 256)).Get(ctx, nil)
+	})
+	if err != nil {
+		return "", err
+	}
+	err = workflow.SetUpdateHandler(ctx, UpdateLocalActivity, func(ctx workflow.Context) error {
+		localActCtx := workflow.WithLocalActivityOptions(ctx, defaultLocalActivityOpts())
+		return workflow.ExecuteLocalActivity(
+			localActCtx, "Payload", activities.MakePayloadInput(0, 256)).Get(ctx, nil)
+	})
+	if err != nil {
+		return "", err
+	}
+	signchan := workflow.GetSignalChannel(ctx, ASignal)
+	workflow.Go(ctx, func(ctx workflow.Context) {
+		for {
+			signchan.Receive(ctx, nil)
+		}
+	})
+
+	for i := params.InitialIteration; i < params.Iterations; i++ {
+		// Repeat the steps as defined by the ancient ritual
+		actCtx := workflow.WithActivityOptions(ctx, defaultActivityOpts())
+		err = workflow.ExecuteActivity(
+			actCtx, "Payload", activities.MakePayloadInput(256, 256)).Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+
+		err = workflow.ExecuteActivity(actCtx, "SelfQuery", "myquery").Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+
+		err = workflow.ExecuteActivity(actCtx, "SelfDescribe").Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+
+		localActCtx := workflow.WithLocalActivityOptions(ctx, defaultLocalActivityOpts())
+		err = workflow.ExecuteLocalActivity(
+			localActCtx, "Payload", activities.MakePayloadInput(0, 256)).Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+		err = workflow.ExecuteLocalActivity(
+			localActCtx, "Payload", activities.MakePayloadInput(0, 256)).Get(ctx, nil)
+		if err != nil {
+			return "", err
+		}
+
+		// Do some stuff in parallel
+		err = RunConcurrently(ctx,
+			func(ctx workflow.Context) error {
+				child := workflow.ExecuteChildWorkflow(ctx, "throughputStressChild")
```

Comment:
The existing throughputstress workflows do not cancel appropriately, which is a bit annoying, but I think it would be a really nice non-breaking departure to handle that here with like `PARENT_CLOSE_POLICY_TERMINATE` or something. What do you think? 

Same might apply to activity cancellation as well if any are long running, though it doesn't look like they are at a glance.

Reply:
Cancels should propagate to all of these by default and work as expected. I'm not sure I want to change the close policy to terminate b/c all the children should conclude rapidly as well and I don't want to cover up something getting stuck with that. I'll be adding visibility checks to that effect.

Reply:
Cool, as long as cancellations propagate, I'm happy!